controllers/microk8sconfig_controller.go

```diff
@@ -397,6 +401,7 @@ func (r *MicroK8sConfigReconciler) handleJoiningWorkerNode(ctx context.Context,
 		JoinToken:        "abcdefgwijklmnopqrstuvwxyzABCDEF",
 		PortOfNodeToJoin: portOfNodeToConnectTo,
 		IPOfNodeToJoin:   ipOfNodeToConnectTo,
+		Version:          *machine.Spec.Version,
```

Comment:
One thing I don't get is why the version comes from the machine spec?

Shouldn't it be set only in the microk8s config CRD?

Reply:
I believe we cannot alter the machine spec [1] as it comes with the clusterapi. I wouldn't duplicate this field in our configs.

[1] https://github.com/kubernetes-sigs/cluster-api/blob/main/api/v1beta1/machine_types.go#L74